include/stl2/detail/iterator/move_iterator.hpp

```diff
@@ -90,6 +91,33 @@ STL2_OPEN_NAMESPACE {
 				static_cast<void>(++current_)
 			)
 
+			// Experimental support for move_iterator post-increment
+			using __post_inc_t = decltype(current_++);
+			Readable{R}
+			struct __proxy {
+				R __tmp;
+				decltype(auto) operator*()
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(__tmp)
+				)
+				friend decltype(auto) iter_move(const __proxy& that)
+				STL2_NOEXCEPT_RETURN(
+					__stl2::iter_move(that.__tmp)
+				)
+			};
+
+			STL2_CONSTEXPR_EXT cursor post_increment()
+			noexcept(noexcept(cursor{cursor{__stl2::exchange(current_, __stl2::next(current_))}}))
+			requires Same<__post_inc_t, I>() {
+				return cursor{__stl2::exchange(current_, __stl2::next(current_))};
```

Comment:
Cool, thanks for the sanity check. Squash-merge or don't, I don't care. I just wanted to make sure I understood the subtleties here.